server/ent/ent.js

```diff
@@ -90,7 +92,133 @@ function nmSearch(node) {
 }
 
 module.exports.strictSearch = strictSearch
-function strictSearch(node, nmInstances = [], globalNmList = []) {
+function strictSearch(node, fasta) {
+	let entResults = strictSearchHelper(node)
+	// Filter out nonmono pairs before returning 
+	let finalResults = {species:entResults.species,nm:[]}
+
+	const fastaData = parseFasta(fasta)
+	// Build a dict map of species -> sequence 
+	let sequenceMap = {}
+	for(let obj of fastaData){
+		sequenceMap[obj.species] = obj.sequence
+	}
+
+	// For each species S marked in an nm pair, 
+		// Go through all other pairs with S in them
+		// Find the shortest pair 
+	// If that pair is shorter than the current pair including either of them 
+	// That is the new shortest pair 
+	let allSpecies = []
+	let speciesCheck = {}
+
+	for (let result of entResults.nm) {
+		let sp1 = result.pair[0]
+		let sp2 = result.pair[1]
+		
+		if(!speciesCheck[sp1.ident]){
+			allSpecies.push(sp1.ident)
+			speciesCheck[sp1.ident] = true
+		}
+		if(!speciesCheck[sp2.ident]){
+			allSpecies.push(sp2.ident)
+			speciesCheck[sp2.ident] = true
+		}
+	}
+
+	let shortestPairs = {}
+
+	for(let speciesIdent of allSpecies){
+		let smallestResult = undefined 
+		let smallestDist = -1
+
+		for (let result of entResults.nm) {
+			let sp1 = result.pair[0]
+			let sp2 = result.pair[1]
+			let id1 = sp1.name + LABEL_DIVIDER + sp1.ident
+			let id2 = sp2.name + LABEL_DIVIDER + sp2.ident
+
+			let sequence1 = sequenceMap[id1]
+			let sequence2 = sequenceMap[id2]
+
+			let dist = hammingDistance(sequence1,sequence2)
+			
+			if(sp1.ident == speciesIdent || sp2.ident == speciesIdent){
+				if(smallestDist == -1){
+					smallestDist = dist 
+				}
+
+				if(dist <= smallestDist){
+					smallestDist = dist 
+					smallestResult = result
+				}
+			}
+
+		}
+
+		let sp1 = smallestResult.pair[0]
+		let sp2 = smallestResult.pair[1]
+		let id1 = sp1.name + LABEL_DIVIDER + sp1.ident
+		let id2 = sp2.name + LABEL_DIVIDER + sp2.ident
+
+		if(    (shortestPairs[id1] == undefined || smallestDist < shortestPairs[id1].dist) 
+			&& (shortestPairs[id2] == undefined || smallestDist < shortestPairs[id2].dist)  
+		   ){
```

Comment:
`npm run prettier`, please.  (This should resolve one of the [failing CI builds](https://circleci.com/gh/hybsearch/hybsearch/531), as well as keep this file in conformance with a general style.)